File: app/models/Note.java
package models;

import com.google.common.base.Joiner;
import com.google.common.base.Splitter;
import com.google.common.collect.Lists;
import play.data.validation.MinSize;
import play.data.validation.Required;
import play.db.jpa.Model;
import play.modules.guice.InjectSupport;
import services.markdown.ContentConverter;

import javax.inject.Inject;
import javax.persistence.Entity;
import javax.persistence.Lob;
import java.util.ArrayList;

/**
 * Date: 16.01.2012 at 21:28
 *
 * @author Marcin Swierczynski
 */
@InjectSupport
@Entity
public class Note extends Model {

	private static final String NEW_LINE_SEPARATOR = "\n";

	@Inject
	static ContentConverter markdownConverter;

	@Required
	@Lob
	@MinSize(5)
	public String content;

	public Note(String content) {
		this.content = content;
	}

	@Override
	public String toString() {
		return getTitle();
	}

	public String getTitle() {
		return splitOnLines().get(0);
	}
	
	public String getContentWithoutTitle() {
		ArrayList<String> noteLines = splitOnLines();
		noteLines.remove(0);
		String contentWithoutTitle = Joiner.on(NEW_LINE_SEPARATOR).join(noteLines);
		return contentWithoutTitle.trim();
	}

	public String getConvertedContent() {
		return markdownConverter.convert(getContentWithoutTitle());
	}

	private ArrayList<String> splitOnLines() {
		return Lists.newArrayList(Splitter.on(NEW_LINE_SEPARATOR).split(content));
	}
}

File: app/models/Note.scala
package models

import play.api.db._
import play.api.Play.current
import anorm._
import anorm.SqlParser._
import services.markdown.ComponentRegistry

/**
 * Date: 17.03.2012 at 14:20
 *
 * @author Marcin Swierczynski
 */

case class Note(id: Long, content: String) {

	val NEW_LINE_SEPARATOR: String = "\n"
	val markdownConverter = ComponentRegistry.markdownConverter

	def title(): String = content.split(NEW_LINE_SEPARATOR)(0)

	def contentWithoutTitle(): String = content.split(NEW_LINE_SEPARATOR).drop(1) mkString NEW_LINE_SEPARATOR

	def convertedContent(): String = markdownConverter.convert(contentWithoutTitle())

}

object Note {

	def all(): List[Note] = DB.withConnection { implicit c =>
		SQL("select * from note").as(note *)
	}

	def create(content: String) {
		DB.withConnection { implicit c =>
			SQL("insert into note (content) values ({content})").on('content -> content).executeUpdate()
		}
	}

	def delete(id: Long) {
		DB.withConnection { implicit c =>
			SQL("delete from note where id = {id}").on('id -> id).executeUpdate()
		}
	}
	
	def findByContent(content: String): List[Note] = DB.withConnection { implicit c =>
		SQL("select * from note where content like '%{content}%'").on('content -> content).as(note *)
	}

	val note = {
		get[Long]("id") ~
		get[String]("content") map {
			case id~label => Note(id, label)
		}
	}
}
